### PR TITLE
Allow setting of playbackRates and playbackRateControls

### PR DESCRIPTION
**Pull request by jnatalzia**
### This PR will...
Allow users to reset playbackRates and playbackRateControls mid-playback

### Why is this Pull Request needed?
The SDKs need to be able to remove unsupported playback rates as well as disable controls until rates are known.

### Are there any points in the code the reviewer needs to double check?
None that I can think of beyond checking to see if I overlooked a simpler solution

### Are there any Pull Requests open in other repos which need to be merged with this?
Nope 

#### Addresses Issue(s):

JW8-2685

### CODE REVIEW

File: src/js/view/controls/settings-menu.js [hunk 1]
```diff
@@ -204,6 +203,10 @@ export function setupSubmenuListeners(settingsMenu, controlbar, viewModel, api)
         }
     }, settingsMenu);
 
+    model.change('playbackRateControls', () => {
```
Comment (robwalch) [inline src/js/view/controls/settings-menu.js]: Does this needs to be invoked immediately? `model.change` is vs `model.on('change:...` which is only invoked after a change to the current value.

File: src/js/view/controls/settings-menu.js [hunk 2]
```diff
@@ -12,11 +12,10 @@ import {
...
 export function createSettingsMenu(controlbar, onVisibility, localization) {
     const settingsButton = controlbar.elements.settingsButton;
-    const onMenuEmpty = () => {
-        settingsButton.hide();
-    };
-
-    const settingsMenu = SettingsMenu(onVisibility, onMenuEmpty, localization);
+    const settingsMenu = SettingsMenu(onVisibility, {
+        hide: () => settingsButton.hide(),
+        show: () => settingsButton.show()
```
Comment (robwalch) [inline src/js/view/controls/settings-menu.js]: Is this hiding all settings or just the rate tab in the settings?

Ideally we only hide/show or add/remove the tab, and the menu button is shown or hidden based on the number of settings tabs.

Comment (robwalch) [reply src/js/view/controls/settings-menu.js]: I get it now.

File: src/js/view/controls/settings-menu.js (hunk 1 above)
Comment (jnatalzia) [reply src/js/view/controls/settings-menu.js]: Ooo nice, will update

**Pull request merged**